internal/cmd/gocdk/launch.go

```diff
@@ -187,6 +209,172 @@ func (local *localLauncher) Launch(ctx context.Context, input *LaunchInput) (*ur
 	return serveURL, nil
 }
 
+// cloudRunLauncher pushes Docker containers to Google Container Registry and
+// creates/updates a Cloud Run service.
+type cloudRunLauncher struct {
+	logger    *log.Logger
+	client    *cloudrun.APIService
+	dockerEnv []string
+	dockerDir string
+}
+
+// Launch implements Launcher.Launch.
+func (crl *cloudRunLauncher) Launch(ctx context.Context, input *LaunchInput) (*url.URL, error) {
+	projectID := specifierStringValue(input.Specifier, "project_id")
+	location := specifierStringValue(input.Specifier, "location")
+	serviceName := specifierStringValue(input.Specifier, "service_name")
+	if projectID == "" || location == "" || serviceName == "" {
+		return nil, xerrors.New("cloud run launch: specifier does include project_id, location, and/or service_name")
```

Comment:
does -> missing ?

Reply:
And maybe be more specific than `specifier`; IIUC, this is the `launch_specifier` section in `outputs.tf`?